### ISSUE TITLE
Navigation Meshes and Path finding

### ISSUE THREAD

**Issue by joshuateveryday** [EasyFarm/EasyFarm#318]
This issue will be used to track the progress of implementing a new navigation system.

Implement Recast/Detour wrapper and path-finding algorithm based on Dark Star project.
DSP has nav meshes available already.

- [x] Implement Recast/Detour Wrapper
- [x] Create Navigation Mesh Handler
- [x] Implement Path-Finding Algorithms
- [x] Re-work Navigation system to use Nav Mesh path finding.

**Comment by AzoRAI** [comment]
@Mykezero Good news! Been a while since I was able to continue work on this system, but since it's necessary for all other farming modes, I picked up. 

I've run into some challenges, but I've been able to get a basic navmesh system into the project.
I have the system loading navmesh and implementing IDisposable so we don't run into memory management issues. I've copied some pathfinding algorithms from other similar projects and the main remaining challenge is to test in a variety of situations using the "grind" mode for main testing.  

This should allow us to simplify, enhance, or do away with the current navigation node systems. I'm still testing what works best in this scenario.   I intend to PR my branch to main repo, but it's based on old code so some update/merge will be required for new system!
